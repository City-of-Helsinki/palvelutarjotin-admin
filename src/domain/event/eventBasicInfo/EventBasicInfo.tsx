import React from 'react';
import { useTranslation } from 'react-i18next';

import TextTitle from '../../../common/components/textTitle/TextTitle';
import { EventQuery } from '../../../generated/graphql';
import { Language } from '../../../types';
import getLocalizedString from '../../../utils/getLocalizedString';
import styles from './eventBasicInfo.module.scss';

type Props = {
  eventData: EventQuery;
  language: Language;
};

const EventBasicInfo: React.FC<Props> = ({ eventData, language }) => {
  const { t } = useTranslation();

  const name = getLocalizedString(eventData.event?.name || {}, language);
  const shortDescription = getLocalizedString(
    eventData.event?.shortDescription || {},
    language
  );
  const description = getLocalizedString(
    eventData.event?.description || {},
    language
  );

  const duration = eventData.event?.pEvent?.duration;
  const neededOccurrences = eventData.event?.pEvent?.neededOccurrences;

  const infoUrl = getLocalizedString(eventData.event?.infoUrl || {}, language);
  const inLanguage = eventData.event?.inLanguage;
  const audience = eventData.event?.audience;

  const keywords = eventData.event?.keywords;

  return (
    <div className={styles.eventBasicInfo}>
      <h2>{t('eventDetails.basicInfo.title')}</h2>
      {name && (
        <>
          <TextTitle>{t('eventDetails.basicInfo.labelName')}</TextTitle>
          <p>{name}</p>
        </>
      )}

      {shortDescription && (
        <>
          <TextTitle>
            {t('eventDetails.basicInfo.labelShortDescription')}
          </TextTitle>
          <p>{shortDescription}</p>
        </>
      )}

      {description && (
        <>
          <TextTitle>{t('eventDetails.basicInfo.labelDescription')}</TextTitle>
          <p>{description}</p>
        </>
      )}

      <TextTitle>{t('eventDetails.basicInfo.labelImage')}</TextTitle>
      <p>TODO</p>

      {infoUrl && (
        <>
          <TextTitle>{t('eventDetails.basicInfo.labelInfoUrl')}</TextTitle>
          <p>{infoUrl}</p>
        </>
      )}

      <div className={styles.durationRow}>
        <div>
          <TextTitle>{t('eventDetails.basicInfo.labelDuration')}</TextTitle>
          <p>{duration}</p>
        </div>
        <div>
          <TextTitle>{t('eventDetails.basicInfo.labelInLanguage')}</TextTitle>
          <p>
            {inLanguage
              ?.map((item) => getLocalizedString(item.name || {}, language))
              .filter((item) => item)
              .sort()
              .join(', ') || '-'}
          </p>
        </div>
        <div>
          <TextTitle>{t('eventDetails.basicInfo.labelAudience')}</TextTitle>
          <p>
            {audience
              ?.map((keyword) =>
                getLocalizedString(keyword.name || {}, language)
              )
              .filter((item) => item)
              .sort()
              .join(', ') || '-'}
          </p>
        </div>
      </div>

      <div className={styles.neededOccurrencesRow}>
        <div>
          <TextTitle>
            {t('eventDetails.basicInfo.labelNeededOccurrences')}
          </TextTitle>
          <p>{neededOccurrences}</p>
        </div>
        <div>
          <TextTitle>Toiminnan tyyppi</TextTitle>
          <p>TODO</p>
        </div>
        <div>
          <TextTitle>VASU / OPS sisällöt</TextTitle>
          <p>TODO</p>
        </div>
      </div>

      <TextTitle>Hinta</TextTitle>
      <p>TODO</p>
      <TextTitle>{t('eventDetails.basicInfo.labelKeywords')}</TextTitle>
      <p>
        {keywords
          ?.map((keyword) => getLocalizedString(keyword.name || {}, language))
          .filter((item) => item)
          .sort()
          .join(', ') || '-'}
      </p>
    </div>
  );
};

export default EventBasicInfo;
