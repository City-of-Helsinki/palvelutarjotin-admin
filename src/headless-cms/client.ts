/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import { ApolloClient, InMemoryCache } from '@apollo/client';
import { relayStylePagination } from '@apollo/client/utilities';

const apolloClient = new ApolloClient({
  uri: import.meta.env.VITE_APP_CMS_URI,
  cache: new InMemoryCache({
    typePolicies: {
      Page: {
        fields: {
          children: relayStylePagination(['where', ['search']]),
        },
      },
    },
  }),
  headers: {
    authorization: '',
  },
});
export default apolloClient;
