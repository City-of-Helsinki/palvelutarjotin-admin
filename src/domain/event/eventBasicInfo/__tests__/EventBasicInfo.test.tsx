import { MockedProvider } from '@apollo/client/testing';
import { render, screen } from '@testing-library/react';
import * as React from 'react';

import { ImageDocument } from '../../../../generated/graphql';
import {
  fakeEvent,
  fakeImage,
  fakeLocalizedObject,
} from '../../../../utils/mockDataUtils';
import EventBasicInfo from '../EventBasicInfo';

const event = fakeEvent({
  name: fakeLocalizedObject('Tapahtuma 13.7.2020'),
  shortDescription: fakeLocalizedObject('Testitapahtuma 13.7.2020'),
  description: fakeLocalizedObject('Tapahtuman kuvaus'),
  infoUrl: fakeLocalizedObject('www.testi.fi'),
  images: [
    fakeImage({
      altText: 'Testikuva',
      id: '48566',
      photographerName: 'Testi Kuvaaja',
    }),
  ],
});

const mocks = [
  {
    request: {
      query: ImageDocument,
      variables: {
        id: '48566',
      },
    },
    result: {
      data: {
        image: fakeImage({
          altText: 'Testikuva',
          id: '48566',
          photographerName: 'Testi Kuvaaja',
        }),
      },
    },
  },
];

it('matches snapshot', async () => {
  const { container } = render(
    <MockedProvider mocks={mocks}>
      <EventBasicInfo eventData={{ event }} language="fi" />
    </MockedProvider>
  );

  await screen.findByAltText('Testikuva');

  expect(container).toMatchSnapshot();
});

it('renders and shows all the event details', async () => {
  render(
    <MockedProvider mocks={mocks}>
      <EventBasicInfo eventData={{ event }} language="fi" />
    </MockedProvider>
  );

  const displayedTexts: string[] = [
    event.description.fi!,
    event.shortDescription.fi!,
    event.pEvent.enrolmentEndDays!.toString(),
  ];

  const localizedTexts = [
    'Lyhyt kuvaus (korkeintaan 160 merkki채)',
    'Tapahtuman perustiedot',
    'Tapahtuman nimi',
    'Kuvaus',
    'Ilmoittautuminen sulkeutuu, p채iv채채',
    'Ilmoittautuminen alkaa',
  ];

  localizedTexts.forEach((text) =>
    expect(screen.getByText(text)).toBeInTheDocument()
  );

  displayedTexts.forEach((text) =>
    expect(screen.getByText(text)).toBeInTheDocument()
  );

  const eventImage = await screen.findByAltText(event.images[0].altText!);

  expect(eventImage).toHaveProperty('src', event.images[0].url);
});
