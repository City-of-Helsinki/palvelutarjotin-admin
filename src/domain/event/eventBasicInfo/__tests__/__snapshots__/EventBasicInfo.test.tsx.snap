// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div>
  <div
    class="eventBasicInfo"
  >
    <h2>
      Tapahtuman perustiedot
    </h2>
    <div
      class="textTitle"
    >
      Tapahtuman nimi
    </div>
    <p>
      Tapahtuma 13.7.2020
    </p>
    <div
      class="textTitle"
    >
      Lyhyt kuvaus (korkeintaan 160 merkkiä)
    </div>
    <p>
      Testitapahtuma 13.7.2020
    </p>
    <div
      class="textTitle"
    >
      Kuvaus
    </div>
    <div
      class="description"
    >
      Tapahtuman kuvaus
    </div>
    <div
      class="textTitle"
    >
      Lisätietojen syöttämisen pakollisuus
    </div>
    <p>
      Lisätietojen syöttäminen ei ole pakollista
    </p>
    <div>
      <div
        class="imagePreview"
      >
        <div
          class="textTitle"
        >
          Tapahtuman kuva
        </div>
        <img
          alt="Testikuva"
          class="image"
          src="https://linkedevents.api.test.hel.ninja/media/images/test.png"
        />
      </div>
      <div
        class="textTitle"
      >
        Kuvan alt-teksti
      </div>
      <p>
        Testikuva
      </p>
      <div
        class="textTitle"
      >
        Valokuvaaja
      </div>
      <p>
        Testi Kuvaaja
      </p>
    </div>
    <div
      class="textTitle"
    >
      WWW-osoite, josta saa lisätietoja tapahtumasta
    </div>
    <p>
      www.testi.fi
    </p>
    <div
      class="durationRow"
    >
      <div>
        <div
          class="textTitle"
        >
          Ilmoittautuminen alkaa
        </div>
        <p>
          13.7.2020 klo 09:00
        </p>
      </div>
      <div>
        <div
          class="textTitle"
        >
          Ilmoittautuminen sulkeutuu, päivää
        </div>
        <p>
          3
        </p>
      </div>
    </div>
  </div>
</div>
`;
